File: core/src/main/java/com/billiamram/main/ExamplePresenter.java
package com.billiamram.main;

import com.billiamram.framework.Presenter;

public class ExamplePresenter extends Presenter<ExampleView> {

    private ExampleModelRepository mExampleModelRepository;

    public ExamplePresenter(ExampleModelRepository exampleModelRepository) {
        mExampleModelRepository = exampleModelRepository;
    }

    @Override
    protected void onViewLoaded(ExampleView view) {
        view.loadItems(mExampleModelRepository.findAll());
    }

    @Override
    protected void onViewUnloaded() {
    }
}

File: core/src/main/java/com/billiamram/main/ExamplePresenter.kt
package com.billiamram.main

import com.billiamram.framework.Presenter

class ExamplePresenter(private val mExampleModelRepository: ExampleModelRepository) : Presenter<ExampleView>() {

    override fun onViewLoaded(view: ExampleView) {
        view.loadItems(mExampleModelRepository.findAll())
    }

    override fun onViewUnloaded() {}
}
